File: KalkJava/Kalk/LogicalResult.java
package Kalk;

import java.util.*;

public class LogicalResult extends Result {
	private ArrayList<ArrayList<Boolean>> matrix = new ArrayList<ArrayList<Boolean>>();
	private ArrayList<Character> columns = new ArrayList<Character>();

	public LogicalResult(ArrayList<ArrayList<Boolean>> matrix, ArrayList<Character> c) {
		super();
		this.matrix = new ArrayList<ArrayList<Boolean>>(matrix);
		this.columns = new ArrayList<Character>(c);
	}
	
	public LogicalResult(LogicalResult l) {
		super();
		this.columns = new ArrayList<Character>(l.columns);
		for(int i = 0; i < l.matrix.size(); i++) {
			matrix.add(new ArrayList<Boolean>());
			for(int j = 0; j < l.matrix.get(i).size(); ++j) {
				matrix.get(i).add(l.matrix.get(i).get(j));
			}
		}
	}
	
	public final ArrayList<Character> getColumnNames() {
		return columns;
	}

	public final ArrayList<ArrayList<Boolean>> getMatrix() {
		return matrix;
	}

	@Override
	public final void stampa() {
		for(int i = 0; i < columns.size(); i++) {
			System.out.print(columns.get(i));
			System.out.print("\t");
		}
		System.out.print("Ris: ");
		System.out.print("\n");
		for(ArrayList<Boolean> i : matrix) {
			for(Boolean j : i) {
				System.out.print(j);
				System.out.print("\t");
			}
			System.out.print("\n");
		}
	}

	public final LogicalResult copyFrom(LogicalResult l) {
		if(this != l) {
			for(Character i : l.columns) {
				columns.add(i);
			}
			for(int i = 0; i < l.matrix.size(); i++) {
				matrix.add(new ArrayList<Boolean>());
				for(int j = 0; j < l.matrix.get(i).size(); ++j) {
					matrix.get(i).add(l.matrix.get(i).get(j));
				}
			}
		}
		return this;
	}

	public double toDouble() {
		double s = 0;
		for(ArrayList<Boolean> l : matrix)
			if(l.get(l.size() - 1))
				++s;
		s = s / matrix.size() * 100;
		return s;
	}
}

File: Kalk/model/logicalresult.cpp
#include "logicalresult.h"

LogicalResult::LogicalResult(const std::vector<std::vector<bool>>& matrix, const std::vector<char>& c) : Result(), matrix(matrix), columns(c)
{
}

LogicalResult::LogicalResult(const LogicalResult& l) : Result(), columns(l.columns)
{
    for (unsigned int i = 0; i < l.matrix.size(); i++) {
        matrix.push_back(std::vector<bool>());
        for(unsigned int j = 0; j < l.matrix[i].size(); ++j)
            matrix[i].push_back(l.matrix[i][j]);
    }
}

LogicalResult& LogicalResult::operator =(const LogicalResult& l)
{
    if(this != &l) {
        for (auto i = l.columns.begin(); i != l.columns.end(); ++i)
            columns.push_back(*i);
        for (unsigned int i = 0; i < l.matrix.size(); i++) {
            matrix.push_back(std::vector<bool>());
            for(unsigned int j = 0; j < l.matrix[i].size(); ++j)
                matrix[i].push_back(l.matrix[i][j]);
        }
    }
    return *this;
}

LogicalResult::operator double() const
{
    double s = 0;
    for (std::vector<std::vector<bool>>::const_iterator i = matrix.begin(); i != matrix.end(); ++i)
        if ((*i)[i->size() - 1]) ++s;
    s = s / matrix.size() * 100;
    return s;
}

std::vector<char> LogicalResult::getColumnNames() const
{
    return columns;
}

std::vector<std::vector<bool>> LogicalResult::getMatrix() const
{
    return matrix;
}

void LogicalResult::stampa() const
{
    for(unsigned int i = 0; i < columns.size(); i++)
        std::cout << columns[i] << "\t";
    std::cout << "Ris: " << std::endl;
    for (std::vector<std::vector<bool>>::const_iterator i = matrix.begin(); i != matrix.end(); ++i) {
        for (std::vector<bool>::const_iterator j = i->begin(); j != i->end(); ++j)
            std::cout << *j << "\t";
        std::cout << std::endl;
    }
}

LogicalResult::~LogicalResult()
{
}
